bayes_opt.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+from scipy.optimize import minimize
+from sklearn.gaussian_process import GaussianProcessRegressor
+from sklearn.gaussian_process.kernels import Matern
+
+
+class BayesianOptimizer(object):
+
+    def __init__(self, feature_meta,
```

Comment:
Ah, hrm, I see. Instead of sampling uniformly we can sample from the data itself, but then you'd need to have all the X's...